clearwater-config-manager.root/usr/share/clearwater/clearwater-config-manager/scripts/validate_json.py

```diff
@@ -101,3 +101,29 @@
 
     print(yaml.dump(temp_dict, default_flow_style=False).replace("u'", "'"))
     sys.exit(1)
+
+
+# Carry out final check, that no duplicated keys are present.
+# This check should be done last, after we have confirmed the json is valid.
+def fail_if_duplicated_keys(json_info):
+    parsed_json = dict(json_info)
+    if (len(parsed_json) != len(json_info)):
+        duplicated_key = ''
+        encountered_keys = []
+        for key in json_info:
+            if key[0] in encountered_keys:
+                duplicated_key = str(key[0])
+            else:
+                encountered_keys.append(key[0])
+        raise KeyError(duplicated_key)
+    else:
+        return parsed_json
+
+
+try:
+    json_file = json.load(open(config_file),
+                          object_pairs_hook=fail_if_duplicated_keys)
+except KeyError as e:
```

Comment:
I don' t think `KeyError` is the appropriate here. From the python [docs](https://docs.python.org/2/library/exceptions.html):

> exception KeyError
Raised when a mapping (dictionary) key is not found in the set of existing keys.

`ValueError` might be better:

> exception ValueError
Raised when a built-in operation or function receives an argument that has the right type but an inappropriate value, and the situation is not described by a more precise exception such as IndexError.

Also, I think the exception should contain the full error message so you can just do `print e.message` (like we do further up in the file).